src/main/java/pl/javastart/task/CustomList.java

```diff
@@ -2,5 +2,66 @@
 
 // Uzupełnij klasę. Możesz ją dowolnie modyfikować.
 // Celem jest, żeby przechodziły testy w src/test/java/[...]/CustomListTest
-public class CustomList {
+import java.util.Arrays;
+
+public class CustomList<T> {
+
+    private int size = 0;
+    private static final int INITIAL_CAPACITY = 2;
+    private Object[] elements;
+
+    public CustomList() {
+        elements = new Object[INITIAL_CAPACITY];
+    }
+
+    public void add(T element) {
+        if (size == elements.length) {
+            ensureCapacity();
+        }
+        elements[size] = element;
+        size++;
+    }
+
+    public void add(int index, T element) {
+        if (size == elements.length) {
+            ensureCapacity();
+        }
+        if (index >= 0 && index < size) {
+            int numberOfElementsToBeCopied = elements.length - (index + 1);
+            System.arraycopy(elements, index, elements, index + 1, numberOfElementsToBeCopied);
+            elements[index] = element;
+            size++;
+        } else if (index == size) {
+            add(element);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public T get(int index) {
+        if (index >= 0 && index <= size) {
+            return (T) elements[index];
+        }
+        throw new ArrayIndexOutOfBoundsException("niewlasciwy index elementu");
+    }
+
+    public int size() {
+        return size;
+    }
+
+    public void remove(int index) {
+        int numberOfElementsToBeCopied = elements.length - (index + 1);
+        System.arraycopy(elements, index + 1, elements, index, numberOfElementsToBeCopied);
+        size--;
+    }
```

Comment:
Jeżeli będziemy usuwali z indeksu wykraczającego poza rozmiar, to powinien być rzucony wyjątek IndexOutOfBounds